fastq_filtrator.py

```diff
@@ -0,0 +1,166 @@
+'''
+Sequences example:
+
+@EAS54_6_R1_2_1_413_324
+CCCTTCTTGTCTTCAGCGTTTCTCC
++
+;;3;;;;;;;;;;;;7;;;;;;;88
+@EAS54_6_R1_2_1_540_792
+TTGGCAGGCCAAGGCCGATGGATCA
++
+;;;;;;;;;;;7;;;;;-;;;3;83
+@EAS54_6_R1_2_1_443_348
+GTTGCTTCTGGCGTGGGTGGGGGGG
++EAS54_6_R1_2_1_443_348
+;;;;;;;;;;;9;7;;.7;393333
+'''
+
+
+class Seq:
+    """ Read representation """
+
+    def __init__(self, label, sequence, qscore):
+        # store values
+        self.label = label
+        self.sequence = sequence
+        self.qscore_str = qscore
+        self.qscore_val = []
+        # translate qscore from chars to int
+        q_sum = 0
+        for ch in self.qscore_str:
+            val = ord(ch) - ord('!')
+            self.qscore_val.append(val)
+            q_sum += val
+        self.qscore_averge = q_sum / len(self.qscore_str)
+        # calculate GC in percent
+        cnt = 0
+        for ch in self.sequence:
+            if ch == 'G' or ch == 'C':
+                cnt += 1
+        self.gc_percent = cnt * 100 / len(self.sequence)
```

Comment:
```suggestion
        for ch in self.sequence.upper():
            if ch == 'G' or ch == 'C':
                cnt += 1
        self.gc_percent = cnt * 100 / len(self.sequence)
```
Хороший вариант подсчёта за 1 проход